client/scala/src/test/scala/ai/verta/blobs/dataset/TestPathBlob.scala

```diff
@@ -0,0 +1,71 @@
+package ai.verta.blobs.dataset
+
+import ai.verta.blobs._
+
+import scala.language.reflectiveCalls
+
+import org.scalatest.FunSuite
+import org.scalatest.Assertions._
+
+class TestPathBlob extends FunSuite {
+  /** Verify that a FileMetadata has correct path and does not have invalid parameter
+   *  @param metadata file's metadata
+   *  @param path file's path (to be checked)
+   */
+  def assertMetadata(metadata: FileMetadata, path: String) = {
+    assert(metadata.path.equals(path))
+    assert(metadata.size > 0)
+    assert(metadata.lastModified > 0)
+    assert(metadata.md5.length > 0)
+  }
+
+  def fixture =
+    new {
+      val workingDir = System.getProperty("user.dir")
+      val testDir = workingDir + "/src/test/scala/ai/verta/blobs/testdir"
+      val testfile = testDir + "/testfile"
+      val testSubdir = testDir + "/testsubdir"
+      val testfile2 = testSubdir + "/testfile2"
+    }
+
+  test("PathBlob should retrieve a file's metadata correctly") {
+    val f = fixture
+    var pathBlob = PathBlob(List(f.testfile))
+
+    assertMetadata(pathBlob.getMetadata(f.testfile).get, f.testfile)
+  }
+
+  test("PathBlob should retrieve multiple files correctly") {
+    val f = fixture
+    var pathBlob = PathBlob(List(f.testfile, f.testfile2))
+
+    assertMetadata(pathBlob.getMetadata(f.testfile).get, f.testfile)
+    assertMetadata(pathBlob.getMetadata(f.testfile2).get, f.testfile2)
+  }
+
+  test("PathBlob should not store directory") {
+    val f = fixture
+    val pathBlob = PathBlob(List(f.testDir, f.testSubdir))
+
+    assert(pathBlob.getMetadata(f.testDir).isEmpty)
+    assert(pathBlob.getMetadata(f.testSubdir).isEmpty)
+    assert(pathBlob.getMetadata(f.testfile).isDefined)
+  }
+
+  test("PathBlob should not store invalid paths, but should not stop execution") {
```

Comment:
That's what the Python client did too. I was trying to salvage the valid ones with all the Try magics, but if it's not worth the troubles then I can just let it die.

Reply:
So this is what I'll do to stop the program: if any of the path is invalid, an exception is thrown. If the user doesn't catch this exception, then the program terminates. This is a runtime problem though, since there is no way to check if a file exists at compile time.

Reply:
"That's what the Python client did too." @convoliution this seems like a bug to me.

Reply:
Oh by "that's what the python did too" what i meant was what you said was what it did.